continuum/task_set.py

```diff
@@ -45,40 +49,70 @@ def add_memory(
         :param t_memory: The associated task ids. If not provided, they will be
                          defaulted to -1.
         """
-        self.x = np.concatenate((self.x, x_memory))
-        self.y = np.concatenate((self.y, y_memory))
+        self._x = np.concatenate((self._x, x_memory))
+        self._y = np.concatenate((self._y, y_memory))
         if t_memory is not None:
-            self.t = np.concatenate((self.t, t_memory))
+            self._t = np.concatenate((self._t, t_memory))
         else:
-            self.t = np.concatenate((self.t, -1 * np.ones(len(x_memory))))
+            self._t = np.concatenate((self._t, -1 * np.ones(len(x_memory))))
 
     def plot(
-        self,
-        path: Union[str, None] = None,
-        title: str = "",
-        nb_per_class: int = 5,
-        shape=None
+            self,
+            path: Union[str, None] = None,
+            title: str = "",
+            nb_samples: int = 100,
+            shape=None
     ) -> None:
         """Plot samples of the current task, useful to check if everything is ok.
 
         :param path: If not None, save on disk at this path.
         :param title: The title of the figure.
-        :param nb_per_class: Amount to sample per class.
+        :param nb_samples: Amount of samples randomly selected.
         :param shape: Shape to resize the image before plotting.
         """
-        plot(self, title=title, path=path, nb_per_class=nb_per_class, shape=shape)
+        plot_samples(self, title=title, path=path, nb_samples=nb_samples, shape=shape)
 
     def __len__(self) -> int:
         """The amount of images in the current task."""
-        return self.x.shape[0]
+        return self._x.shape[0]
+
+    def rand_samples(self, nb_samples):
+
+        nb_tot_samples = self._x.shape[0]
+        indexes = np.random.randint(0, nb_tot_samples, nb_samples)
+        return self.get_samples_from_ind(indexes)
+
+    def get_samples_from_ind(self, indexes):
+        batch = torch.zeros(0)
+        labels = torch.zeros(0)
+        task_id = torch.zeros(0)
+
+        for i, ind in enumerate(indexes):
+            # we need to use get item to have the transform used
+            img, y, t = self.__getitem__(ind)
+
+            if i == 0:
+                if len(list(img.shape)) == 2:
+                    size_image = [1] + list(img.shape)
+                else:
+                    size_image = list(img.shape)
+                batch = torch.zeros(([len(indexes)] + size_image))
```

Comment:
If I recall right, it's very slow to grow a numpy or torch tensor likewise. Better aggregate them in a python list and call `torch.cat(THE_LIST)` as the end. (or `vstack` or `tensor` I forgot).

Plus this method will be way shorter likewise.